lib/jellyfish/component.ex

```diff
@@ -0,0 +1,71 @@
+defmodule Jellyfish.Component do
+  @moduledoc """
+  Component is a server side entity that can publishes a track or subscribes to tracks and process them.
+
+  Examples of components are:
+    * FileReader which reads a track from a file.
+    * HLSComponent which saves received tracks to HLS stream.
+  """
+
+  alias Membrane.RTC.Engine.Endpoint.HLS
+
+  @enforce_keys [
+    :id,
+    :type
+  ]
+  defstruct @enforce_keys ++ [:engine_endpoint]
+
+  @type id :: String.t()
+  @type component_type :: :file_reader | :hls
+
+  @typedoc """
+  This module contains:
+  * `id` - peer id
+  * `component_type` - type of component
+  * `engine_endpoint` - engine endpoint for this component
+  """
+  @type t :: %__MODULE__{
+          id: id,
+          type: component_type(),
+          engine_endpoint: struct() | nil
+        }
+
+  @spec new(component_type :: atom()) :: t()
+  def new(component_type) do
+    %__MODULE__{
+      id: UUID.uuid4(),
+      type: component_type
+    }
+  end
+
+  @spec validate_component_type(component_type :: String.t()) ::
+          {:ok, component_type()} | :error
+  def validate_component_type(component_type) do
```

Comment:
This breaks SRP a little bit as not only it validates but also converts

Reply:
I think this is more bad naming and this function should be `parse_component_type`, also function can do multiple things and is not against SRP e.g: `getAndIncrement` in `AtomicInteger` in Java.